src/artifactorypurge/artifactorypurge.py

```diff
@@ -1,64 +1,78 @@
+@click.option(
+    '--dryrun/--nodryrun', default=True, is_flag=True, help='Dryrun does not delete any artifacts. On by default')
+@click.option('--plugin-path', required=True, help='Path to plugin directory')
+@click.option('--default/--no-default', default=True, is_flag=True, help='If false, does not apply default policy')
 #@click.argument('url')
-def purge(dryrun, reponame, project): #, url):
-    exceptions = {}
-
+def purge(dryrun, plugin_path, default):  #, url):
     credentials = load_credentials()
-    artifactory = Artifactory(credentials['artifactory_url'], 
-                              credentials['artifactory_username'],
+    artifactory = Artifactory(credentials['artifactory_url'], credentials['artifactory_username'],
                               credentials['artifactory_password'])
 
-    before = artifactory.list(reponame)
+    plugin_source = setup_pluginbase(plugin_path)
+    before = artifactory.list(None)
     for repo, info in before.items():
-        fp = 0
+        plugin_name = repo.replace("-", "_")
         try:
-            modulename = repo.replace("-", "_")
-            fp, pathname, description = imp.find_module(modulename, ["repositories",])
-            module =  imp.load_module(modulename, fp, pathname, description)
-            artifacts = module.purgelist(artifactory, repo, project)
-            count = artifactory.purge(repo, dryrun, artifacts)
-            LOG.info("processed {}, purged {}".format(repo, count))
-        except IndexError as e:  # FIXME: return to generic catch
-            exceptions[repo] = str(e)
-        except ImportError:
-            continue
-        finally:
-            if fp:
-                fp.close()
+            artifactory_plugin = plugin_source.load_plugin(plugin_name)
+        except ModuleNotFoundError:
+            if default:
+                LOG.info("No plugin found for %s. Applying Default", repo)
+                artifactory_plugin = plugin_source.load_plugin('default')
+            else:
+                LOG.info("No plugin found for %s. Skipping Default", repo)
+                continue
+        artifacts = artifactory_plugin.purgelist(
+            artifactory,
+            repo,
+            None,
+        )
+        count = artifactory.purge(repo, dryrun, artifacts)
+        LOG.info("processed {}, purged {}".format(repo, count))
 
     LOG.info("")
     LOG.info("Purging Performance:")
-    after = artifactory.list(reponame)
+    after = artifactory.list(None)
     for repo, info in after.items():
         try:
             get_performance_report(repo, before[repo], info)
         except IndexError:
             pass
 
-    if len(exceptions):
-        LOG.error("There were errors:")
-        for repo, exception in exceptions.items():
-            LOG.error("  {}: {}".format(repo, exception))
-
-        exit(1)
-
     LOG.info("Done.")
 
     exit(0)
 
 
+def setup_pluginbase(plugin_path):
```

Comment:
This should be called `extra_plugin_path` and probably have a default value.
I also think this should be optional.

Reply:
agreed